### PR TITLE
UIIN-1290 Allow the selection of remote storage locations

### PR DESCRIPTION
**Pull request by uladislausamets**
## Purpose

https://issues.folio.org/browse/UIIN-1290
## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.
- [ ] I've added appropriate record to the CHANGELOG.md
- Does this PR meet or exceed the expected quality standards?
  - [ ] Code coverage on new code is 80% or greater
  - [ ] Duplications on new code is 3% or less
  - [ ] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] If any API-related changes - okapi interfaces and permissions are reviewed/changed correspondingly
  - [ ] There are no breaking changes in this PR.

If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/RemoteStorage/Provider.js [hunk 1]
```diff
@@ -31,7 +31,14 @@ const Provider = ({ resources, mutator, ...rest }) => {
   const checkMoveFromRemoteToNonRemote =
     ({ fromLocationId, toLocationId }) => (fromLocationId in remoteMap) && !(toLocationId in remoteMap);
 
-  const context = { checkMoveFromRemoteToNonRemote };
+  const checkIsRemoteLocation = (locationId) => {
+    if (locationId in remoteMap) {
+      return '(Remote)';
```
Comment (NikitaSedyx) [inline src/RemoteStorage/Provider.js:36]: it's not translated

File: src/RemoteStorage/Provider.js [hunk 2]
```diff
@@ -31,7 +31,14 @@ const Provider = ({ resources, mutator, ...rest }) => {
   const checkMoveFromRemoteToNonRemote =
     ({ fromLocationId, toLocationId }) => (fromLocationId in remoteMap) && !(toLocationId in remoteMap);
 
-  const context = { checkMoveFromRemoteToNonRemote };
+  const checkIsRemoteLocation = (locationId) => {
```
Comment (NikitaSedyx) [inline src/RemoteStorage/Provider.js:34]: for me check is smth boolean

File: src/RemoteStorage/Provider.js [hunk 3]
```diff
@@ -1,57 +1,21 @@
...
   createContext,
   useContext,
-  useEffect,
-  useMemo,
-  useState,
 } from 'react';
-import PropTypes from 'prop-types';
 
-import { stripesConnect } from '@folio/stripes/core';
+import { useRemoteStorageMappings } from '@folio/stripes-smart-components/lib/utils';
```
Comment (aliaksei-chumakou) [inline src/RemoteStorage/Provider.js:6]: I don't think we work that way with SSC. Usually we export required things there in index.js and import them `from '@folio/stripes/smart-components';`

**Review by axelhunn — APPROVED** [review]

File: src/RemoteStorage/Provider.js (hunk 3 above)
Comment (aliaksei-chumakou) [reply src/RemoteStorage/Provider.js]: it's still there. Here you can read details https://github.com/folio-org/stripes/blob/master/doc/stripes-framework.md#migrating-to-v10

File: src/RemoteStorage/Provider.js [hunk 4]
```diff
@@ -1,57 +1,21 @@
...
   createContext,
   useContext,
-  useEffect,
-  useMemo,
-  useState,
 } from 'react';
-import PropTypes from 'prop-types';
 
-import { stripesConnect } from '@folio/stripes/core';
+import { useRemoteStorageMappings } from '@folio/stripes-smart-components';
```
Comment (axelhunn) [inline src/RemoteStorage/Provider.js:6]:
```suggestion
import { useRemoteStorageMappings } from '@folio/stripes/smart-components';
```

**Review by aliaksei-chumakou — APPROVED** [review]

**Pull request merged**